apps/contracts/contracts/Zengo.sol

```diff
@@ -15,13 +15,33 @@ contract ZengoDAO is PermissionsEnumerable, ContractMetadata {
     address public votingTokenAddress; // Address of the ERC20 voting token
     address public deployer;
 
+    // Modify the Design to facilitate current implementation
     struct Proposal {
-        uint256 id;
+        uint256 proposalId;
         string title;
-        string description;
-        address submitter;
+        string proposalDescription;
+        string proposalType;
+        address proposer;
+        Evidence proposerEvidence;
+        Vote[] votingIterations;
+        // requires funding
+        bool isEligibleForFunding;
         bool isVerified;
-        uint256 votes;
+    }
+
+    struct Vote {
+        uint256 proposalId;
+        uint256 totalVotes;
+        mapping (address => bool) vote; 
+        Evidence[] evidences;
+        string[] options; // should this be enums?
```

Comment:
> should this be enums?

For the sake of provide my input, this answer (yes or no) relies on if the FE will pass the options for each Iteration Voting.
Defining the enums will require more effort and perhaps we can define them along the way organically, and then the contract can be updated?
Also using this array of strings give us more flexibility at this point.

just my opinion

Reply:
I shall change the mapping as well for the vote